### PR TITLE
Updates for Monterey Xcode & CLT

### PR DESCRIPTION
**Pull request by fxcoudert**
Following up on partial support in https://github.com/Homebrew/brew/pull/11504

This is `do not merge` for now as I'm waiting for the installation of CLT to check the exact clang version numbers.

### CODE REVIEW

**Comment by issyl0** [comment]
All I think of with this name is 🧀, not ⛰️. Thanks for getting the ball rolling on this, FX!

**Review by MikeMcQuaid — DISMISSED** [review]

**Comment by carlocab** [comment]
If there's a `/Library/Developer/CommandLineTools/SDKs/MacOSX12.sdk` I think we may want to use that.

**Comment by fxcoudert** [comment]
@carlocab what is the rationale for using it on 12 but not on 11? `/Library/Developer/CommandLineTools/SDKs/MacOSX11.sdk` exists on Big Sur

**Comment by carlocab** [comment]
@fxcoudert There isn't one. I believe the plan is to start using that too: https://github.com/Homebrew/brew/pull/10769

We switched to the unversioned one because there wasn't a `MacOSX11.sdk` at the time we made the switch.

**Comment by Bo98** [comment]
Yeah we plan to use MacOSX11.sdk soon. It requires bumping the minimum CLT version to 11.5 so we needed to make sure most people had the chance to update first. Previous CLT versions didn't have it.

We're probably near the point now where we can merge that PR.

File: Library/Homebrew/os/mac/pkgconfig/12/libcurl.pc [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+exec_prefix=/usr
+libdir=${exec_prefix}/lib
+includedir=${prefix}/include
+supported_protocols="DICT FILE FTP FTPS GOPHER HTTP HTTPS IMAP IMAPS LDAP LDAPS POP3 POP3S RTSP SMB SMBS SMTP SMTPS TELNET TFTP"
+supported_features="AsynchDNS IPv6 Largefile GSS-API Kerberos SPNEGO MultiSSL NTLM NTLM_WB SSL libz HTTP2 UnixSockets HTTPS-proxy"
+
+Name: libcurl
+URL: https://curl.haxx.se/
+Description: Library to transfer files with ftp, http, etc.
+Version: 7.64.1
```
Comment (Bo98) [inline Library/Homebrew/os/mac/pkgconfig/12/libcurl.pc:37]:
```suggestion
supported_protocols="DICT FILE FTP FTPS GOPHER HTTP HTTPS IMAP IMAPS LDAP LDAPS MQTT POP3 POP3S RTSP SMB SMBS SMTP SMTPS TELNET TFTP"
supported_features="AsynchDNS GSS-API HTTPS-proxy IPv6 Kerberos Largefile MultiSSL NTLM NTLM_WB SPNEGO SSL UnixSockets alt-svc libz"

Name: libcurl
URL: https://curl.se/
Description: Library to transfer files with ftp, http, etc.
Version: 7.76.0
```

(Side note: a bit odd they removed HTTP2 support?)

File: Library/Homebrew/os/mac/pkgconfig/12/sqlite3.pc [hunk 2]
```diff
@@ -0,0 +1,12 @@
+homebrew_sdkroot=/Library/Developer/CommandLineTools/SDKs/MacOSX12.sdk
+prefix=${homebrew_sdkroot}/usr
+exec_prefix=/usr
+libdir=${exec_prefix}/lib
+includedir=${prefix}/include
+
+Name: SQLite
+Description: SQL database engine
+Version: 3.32.3
```
Comment (Bo98) [inline Library/Homebrew/os/mac/pkgconfig/12/sqlite3.pc:9]:
```suggestion
Version: 3.35.3
```

File: Library/Homebrew/os/mac/pkgconfig/12/libcurl.pc (hunk 1 above)
Comment (chrisfinazzo) [reply Library/Homebrew/os/mac/pkgconfig/12/libcurl.pc:37]: Almost all major browsers have [support][] now, even if it is still a draft according to IETF. It's behind a flag if you're running Big Sur, but it's been in the Technology Preview for a while and could change with Monterey.

(Not running it - just a guess)

A few libraries and servers are still working on support - notably libcurl - but most seem to be ready.

[support]: https://caniuse.com/http3

Comment (Bo98) [reply Library/Homebrew/os/mac/pkgconfig/12/libcurl.pc:37]: Yeah I get that, but it was present as a "supported feature" in Curl on Big Sur but appears to have disappeared entirely when I checked the Monterey SDK.

(Comparing `curl-config --features` between 11.3 and 12.0 SDKs)

Another sign was `otool -L /usr/lib/libcurl.4.dylib` linking to `/usr/lib/libapple_nghttp2.dylib` but I don't have a 12.0 install to check.

Comment (kidonng) [reply Library/Homebrew/os/mac/pkgconfig/12/libcurl.pc:37]: I remember curl does have HTTP2 feature in the new CLT on macOS 12. But now I don't have access to it anymore so I cannot check.

**Pull request merged**